packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -72,27 +73,66 @@ export const styles = theme => ({
   },
 });
 
-class Tooltip extends React.Component {
-  ignoreNonTouchEvents = false;
-
-  constructor(props) {
-    super();
-    this.isControlled = props.open != null;
-    this.state = {
-      open: null,
+function useMountedRef() {
+  const mountedRef = React.useRef(false);
+  React.useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
     };
+  }, []);
 
-    if (!this.isControlled) {
-      // not controlled, use internal state
-      this.state.open = false;
-    }
-  }
+  return mountedRef;
+}
 
-  componentDidMount() {
+function Tooltip(props) {
+  const {
+    children,
+    classes,
+    disableFocusListener,
+    disableHoverListener,
+    disableTouchListener,
+    enterDelay,
+    enterTouchDelay,
+    id,
+    interactive,
+    leaveDelay,
+    leaveTouchDelay,
+    onClose,
+    onOpen,
+    open: openProp,
+    placement,
+    PopperProps,
+    theme,
+    title,
+    TransitionComponent,
+    TransitionProps,
+    ...other
+  } = props;
+
+  const [openState, setOpenState] = React.useState(false);
+  const [, forceUpdate] = React.useState(0);
+  const ignoreNonTouchEvents = React.useRef(false);
+  const { current: isControlled } = React.useRef(props.open != null);
+  const childrenRef = React.useRef();
+  // can be removed once we drop support for non ref forwarding class components
+  const handleOwnRef = React.useCallback(ref => {
+    childrenRef.current = ReactDOM.findDOMNode(ref);
```

Comment:
At this point we should just put a `childNode` into state and update that instead so we don't have to use forceUpdate hacks. This could also reduce the amount of renders since React would bail out if the node didn't change.